client/services/FeedbackGeneratorService.js

```diff
@@ -30,6 +30,33 @@ tie.factory('FeedbackGeneratorService', [
     RUNTIME_ERROR_FEEDBACK_MESSAGES, WRONG_LANGUAGE_ERRORS, LANGUAGE_PYTHON,
     CLASS_NAME_AUXILIARY_CODE, CLASS_NAME_SYSTEM_CODE, PARAGRAPH_TYPE_TEXT,
     PARAGRAPH_TYPE_CODE, PARAGRAPH_TYPE_SYNTAX_ERROR) {
+    /**
+     *
+     */
+    var UNFAMILIARITY_THRESHOLD = 5;
+
+    var consecutiveSyntaxErrorCounter = 0;
+    var consecutiveSameErrorCounter = 0;
+    var consecutiveWrongLanguageCounter = 0;
+    var previousErrorString = '';
+
+    /**
+     *
+     */
+    var _resetCounters = function(currentCtr) {
+      var notResetCtr = currentCtr || '';
```

Comment:
Perhaps a short comment explaining the purpose for checking if currentCtr is null?